Tropos/Models/Temperature.swift

```diff
@@ -0,0 +1,46 @@
+import Foundation
+
+@objc enum TemperatureComparison: Int {
+    case Same, Hotter, Warmer, Cooler, Colder
+}
+
+private enum TemperatureLimit: Int {
+    case Hotter = 32
+    case Colder = 75
+}
+
+@objc class Temperature: NSObject {
+    let fahrenheitValue: Int
+    
+    lazy var celsiusValue: Int = {
+        return Int(round(Float(self.fahrenheitValue - 32) * 5.0 / 9.0))
+    }()
+    
+    init(fahrenheitValue: Int) {
+        self.fahrenheitValue = fahrenheitValue
+    }
+    
+    func temperatureDifferenceFrom(temperature: Temperature) -> Temperature {
+        return Temperature(fahrenheitValue: fahrenheitDifference(temperature))
+    }
+
+    func comparedTo(temperature: Temperature) -> TemperatureComparison {
+        let diff = fahrenheitDifference(temperature)
```

Comment:
This will just remove `fahrenheitDifference` and we can use overloaded operator in `temperatureDifferenceFrom`